doc/rst/technotes/dsi.rst

```diff
@@ -954,5 +954,173 @@ implicitly for each ``GlobalDomain`` object and so will be local in any case.
 Phase 4: Bulk-Transfer Interface
 =================================
 
-The bulk-transfer interface design is still in flux. Once finalized, it will
-be documented here.
+What is the bulk-transfer interface?
+------------------------------------
+
+While the Chapel language does not require a specific implementation for array
+assignment, in practice the values are assigned individually:
+
+ .. code-block:: chapel
+
+  for (d, s) in zip(dest, src) {
+    d = s;
+  }
+
+When assignment occurs between arrays on different locales, this can result in
+significant communication overhead. The bulk-transfer interface aims at
+reducing communication between locales by enabling domain maps to handle
+certain kinds of assignments themselves.
+
+How does the bulk-transfer interface work with DSI?
+---------------------------------------------------
+
+The bulk-transfer interface is an optional interface supported only on array
+descriptors. The following methods are prefixed with ``doi`` for "Domain map
+Optional Interface." An array descriptor can implement one or more of these
+methods to opt-in to support for bulk-transfers:
+
+.. method:: proc GlobalArray.doiBulkTransferToKnown(myDom:domain, otherClass, otherDom:domain) : bool
+
+  Perform an array assignment from ``this`` to ``otherClass``.
+
+.. method:: proc GlobalArray.doiBulkTransferFromKnown(myDom:domain, otherClass, otherDom:domain) : bool
+
+  Perform an array assignment from ``otherClass`` to ``this``
+
+.. method:: proc GlobalArray.doiBulkTransferToAny(myDom:domain, otherClass, otherDom:domain) : bool
+
+  Perform an array assignment from ``this`` to ``otherClass``.
+
+.. method:: proc GlobalArray.doiBulkTransferFromAny(myDom:domain, otherClass, otherDom:domain) : bool
+
+  Perform an array assignment from ``otherClass`` to ``this``
+
+Each method shares these arguments:
+
+* ``myDom:domain`` - the indices used when transferring to/from ``this``
+
+* ``otherClass`` - a class inheriting from ``BaseArr`` that represents the
```

Comment:
Is BaseArr documented explicitly, such that we could link to it?  I think someone reading this document would try to search for it, so it would be better to give them a link or maybe be more explicit about where they would find it.

Reply:
BaseArr is not currently documented. It is briefly introduced in an earlier section of this technote, but I don't think we can say much more until we have time to nail down more of DSI.

Reply:
Should we make it more vague for now, then?

Reply:
Perhaps unfortunately, inheriting from ``BaseArr`` is a requirement for domain maps at this time so I think we should mention it. This document in general is still in a fair amount of flux, so I'm comfortable leaving this line as-is.

Reply:
Seems fair to me!